<a href="https://colab.research.google.com/github/Isokonji/sample-test/blob/main/one_two_paired_sample_t_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
from scipy import stats

calculate the resting systolic blood pressure of 20 first year resident female doctors and compare it to the general population mean of 120 mmHg

the null hypothesis
there is no significant difference betweebthe blood pressure of resident doctors and general polpulation

the alternate hypothesis
there is a statistically significant difference between the blood pressures of resident femal doctors and the general population

In [2]:
#sample of blood pressure of 20 female docts
female_doctor_bps= [128,127,118,115,114,142,133,140,132
                    ,131,111,132,149,122,139,136,129,126,128,119]
stats.ttest_1samp(female_doctor_bps,120)

TtestResult(statistic=np.float64(3.8207113134189), pvalue=np.float64(0.0011539093469827148), df=np.int64(19))

our t-statistics value is 3.820 and along with the degree of freedom being 19 which is n-1, this can be used to calculate the p-value , here p-value is 0.0012 which is far less than the threshold of 0.05 or 0.01, so we reject the null hypothesis and can say there is statistically significant difference between the resting systolic blood pressure of the resident female doctor and the general population

TWO SAMPLES T TEST


here we will compare the blood pressure of male consultant doctors with the junior resident female doctors we explored above

null hypothesis
there is no signioficant difference betweenb thge blood pressures of make consultant doctors and junior resident female doctors

alternate hypotheis
there is a stattistically significant difference between the blood pressure of male consultant doctors and junior resident female doctors

In [5]:
male_doctors_bps= [118,115,112,120,124,130,123,110,120,121,123,
                 125,129,130,112,117,119,120,123,128]
male_doctors_bps

stats.ttest_ind(female_doctor_bps,male_doctors_bps)

TtestResult(statistic=np.float64(2.924728298668045), pvalue=np.float64(0.0057855078565290236), df=np.float64(38.0))

our t-statistics value is 2.92 and degree of freefdom is 38, we use this to calculate the p value , which is 0.005, and since this is less than 0.05 or0.01 we reject the null hypothesis and we can statistically say that there is a significant difference between the resting systolic blood pressure of the resident female doctors and male consultant doctors

PAIRED SAMPLE TEST

Here we want to compare or measure the amount of sleep got by patient before and after taking soporific durgs to help them sleep


null hypothesis

soporic drug has no effect on the sleep duration of the patient

alternate hypothesis

soporific drugs has an effect on the sleep duration of the patient

In [3]:
control=[8.0, 7.1,6.5,6.7,7.2,5.4,4.7,8.1,6.3,4.8]

treatment=[9.9,7.9,7.6,6.8,7.1,9.9,10.5,9.7,10.9,8.2]

stats.ttest_rel(control,treatment)

TtestResult(statistic=np.float64(-3.6244859951782136), pvalue=np.float64(0.0055329408161001415), df=np.int64(9))

out t statsitics value is -1.65 and degree of freedom is 9 this can be used to calculate a p-value.

the p value is 0.055 which is below the 0.05 or 0.01 threshold , so we reject trhe null hypothesis we can say there is a statisical difference in sleep duration caused by soporific drug